integration/tests/eks_connector/eks_connector_test.go

```diff
@@ -130,8 +140,28 @@ var _ = Describe("(Integration) [EKS Connector test]", func() {
 	})
 })
 
+func makeConnectedClusterName() string {
+	return fmt.Sprintf("connected-%s", params.ClusterName)
+}
+
+func deregisterCluster() {
+	connectedClusterName := makeConnectedClusterName()
+	cmd := params.EksctlDeregisterCmd.WithArgs("cluster").
+		WithArgs("--name", makeConnectedClusterName())
+
+	session := cmd.Run()
+	if session.ExitCode() == 0 {
+		fmt.Fprintf(GinkgoWriter, "cleaned up registered cluster %q successfully", connectedClusterName)
+	} else {
+		fmt.Fprintf(GinkgoWriter, "warning: failed to deregister cluster %q; this can be ignored if the test ran successfully to completion", connectedClusterName)
+	}
+}
+
 var _ = AfterSuite(func() {
-	params.DeleteClusters()
+	if !params.SkipCreate && !params.SkipDelete {
+		params.DeleteClusters()
+	}
+	deregisterCluster()
```

Comment:
do we want to test the deregister in an actual `Step` rather than in the AfterEach?